### ISSUE TITLE
Performance curves show very low number of deliveries and 0 losses

### ISSUE THREAD

**Issue by davidpadg** [awjuliani/dfp#7]
Hi Arthur, 
Thank you for this article and code! I'm running the notebook on Windows, with Anaconda python 3.5. My performance curves look nothing like yours, showing zero losses and average of < 1 delivery per episode. I get two warnings on startup but no other errors. In the last cell of the notebook I see this output: 
Starting worker 0
Starting worker 1
Starting worker 2
Starting worker 3
Starting worker 4
Starting worker 5
Starting worker 6
Starting worker 7
C:\Users\dp\AppData\Local\Continuum\Anaconda3\envs\tensor35\lib\site-packages\numpy\core\fromnumeric.py:2889: RuntimeWarning: Mean of empty slice.
  out=out, **kwargs)
C:\Users\dp\AppData\Local\Continuum\Anaconda3\envs\tensor35\lib\site-packages\numpy\core\_methods.py:80: RuntimeWarning: invalid value encountered in double_scalars
  ret = ret.dtype.type(ret / rcount)

![image](https://user-images.githubusercontent.com/5048828/29683039-b4e54784-88d2-11e7-9669-71071f9d6492.png)

![image](https://user-images.githubusercontent.com/5048828/29683142-f441d352-88d2-11e7-874e-9ee4c0e88cc2.png)

any suggestions on what might be wrong and how to correct it would be much appreciated!

**Comment by Mazecreator** [comment]
Hi,

Just a guess, but is is possible the m = m1[:] in the "Work()" folder may need to be a NP.COPY:
                    m = np.copy(m1)

Try changing that line.  I can't get the original code to run, so m1 may not be a list that needs to be copied.

**Comment by davidpadg** [comment]
Changing m=m1[:] to m=np.copy(m1) has no effect. 
It turns out the warnings were due to this line: mean_value = np.mean(self.episode_mean_values[-50:])
the only other time episode_mean_values is referenced is in the init of the worker class: self.episode_mean_values = []  - so I'd think everyone would get a warning like this and ignores it. 

That warning was not the real problem however, I also updated all the calls to zip to be compatible with python 3. list(zip(xx))  - which seemed to fix the problem
